contributors/devel/sig-testing/e2e-tests.md

```diff
@@ -587,9 +595,67 @@ refine requirements:
     - `[Alpha]`: This is a test that exercises an alpha feature
     - `[Beta]`: This is a test that exercises a beta feature
 
+    Conceptually, these are non-default requirements as defined above under
+    `[Feature:.+]`, but for historic reasons and the sake of brevity they don't
+    have that prefix when embedded in test names. They *do* have that prefix in the
+    Ginkgo v2 label, so use e.g. `--filter-label=Feature:Alpha`.
```

Comment:
My intent was to also add `Feature:Privileged` and `Feature:Internet` as Ginkgo labels. But when looking at the code I found that `[Internet]` isn't in the code base at all (should be removed above?) and `[Privileged]` is only used here:
```
test/e2e/network/conntrack.go:  ginkgo.It("proxy implementation should not be vulnerable to the invalid conntrack state bug [Privileged]", func(ctx context.Context) {
test/e2e/network/kube_proxy.go: ginkgo.It("should set TCP CLOSE_WAIT timeout [Privileged]", func(ctx context.Context) {
```

I don't see any Prow job which uses it. Shall we remove it above and in the code?

Reply:
`\[Deprecated\]` is used in some jobs, but not in any current test. It probably shouldn't become a `Feature:Deprecated`, that doesn't match the intent. That means that I need to update this new paragraph a bit so that it is clear that it only refers to `[Alpha]` and `[Beta]`.

Reply:
`Feature:Internet` means the cluster needs Internet access, is common to run CI on airgapped environment and this test broke those CIs

Reply:
but if we prefix just with a different prefix we get this simpler, no? instead of reusing Feature we can use Graduation or Stability

```
Graduation:Alpha
Stability:Beta

```

Reply:
See [below](https://github.com/kubernetes/community/pull/7824#pullrequestreview-2006926630).

Reply:
> Feature:Internet means the cluster needs Internet access, is common to run CI on airgapped environment and this test broke those CIs, these CIs need to be able to filter.

But we don't use it anymore?

Reply:
AFAIK we never used `[Feature:Internet]`; we have:
- `[Feature:Networking-IPv4]`: requires access to external IPv4 addresses
- `[Feature:Networking-IPv6]`: requires access to external IPv6 addresses
- `[Feature:Networking-DNS]`: requires the ability to resolve (but _not_ connect to) the IPs of external domain names

Though in #124422 I had to tweak a test to use `curl -4` so that I could then mark it `Networking-IPv4`, so maybe it would be better to have just `Internet` and have it have mean "it's assumed that if pods have an IPv4 `podIPs` value then they have access to the IPv4 Internet and if they have an IPv6 `podIPs` value then they have access to the IPv6 Internet".

Reply:
oh, my mind assumed we already did change it, I remember it was discussed https://github.com/kubernetes/kubernetes/pull/102963#discussion_r654947897

Reply:
So we agree to remove the stale documentation of `[Internet]` here, right?

We don't need to document all the `[Feature:<something>]` here, that's better handled via the source code which gets linked to.

Reply:
yes

Reply:
`[Internet]` removed and examples for `Feature:` definition updated based on real examples from https://github.com/kubernetes/kubernetes/pull/124423 to make it clearer that the non-default requirement could also be about the behavior of the test. The `[Performance]` example was already, but without explaining it's meaning. 

See https://github.com/kubernetes/community/compare/a03958027055983e9d87cc39bd63004f3c6f2474..2c556fa411706c6d151789a98356738058b0ba27